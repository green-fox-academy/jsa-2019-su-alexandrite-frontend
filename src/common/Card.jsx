import React from 'react';
import { View, Text } from 'react-native';
import PropTypes from 'prop-types';
import styles from './styles';

const Card = ({ children, title }) => (
  <View style={styles.card}>
    {title && <Text style={styles.cardTitle}>{title}</Text>}
    {children}
  </View>
);


Card.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node),
  ]),
  title: PropTypes.string,
};

Card.defaultProps = {
  children: [],
  title: undefined,
};

export default Card;
