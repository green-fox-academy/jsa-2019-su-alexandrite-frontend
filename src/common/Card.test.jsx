import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';
import Card from './Card';

test('Card render properly', () => {
  const tree = renderer.create(
    <Card />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Card render properly with title', () => {
  const tree = renderer.create(
    <Card title="test title" />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Card render properly with children', () => {
  const tree = renderer.create(
    <Card><Text>test children</Text></Card>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
