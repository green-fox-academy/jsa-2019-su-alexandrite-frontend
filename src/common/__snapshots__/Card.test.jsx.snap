// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card rednder properly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 6,
        "elevation": 3,
        "marginBottom": 15,
        "padding": 15,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "x": 0,
          "y": 4,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 6,
      },
      null,
    ]
  }
/>
`;

exports[`Card render properly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 6,
        "elevation": 3,
        "marginBottom": 15,
        "padding": 15,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "x": 0,
          "y": 4,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 6,
      },
      null,
    ]
  }
/>
`;

exports[`Card render properly with children 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 6,
        "elevation": 3,
        "marginBottom": 15,
        "padding": 15,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "x": 0,
          "y": 4,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 6,
      },
      null,
    ]
  }
>
  <Text>
    test children
  </Text>
</View>
`;

exports[`Card render properly with title 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 6,
        "elevation": 3,
        "marginBottom": 15,
        "padding": 15,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "x": 0,
          "y": 4,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 6,
      },
      null,
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#333",
        "fontSize": 13,
        "fontWeight": "600",
        "lineHeight": 18,
        "marginBottom": 15,
      }
    }
  >
    test title
  </Text>
</View>
`;
